pkg/odo/cli/catalog/describe/component.go

```diff
@@ -0,0 +1,182 @@
+package describe
+
+import (
+	"fmt"
+	"os"
+	"text/tabwriter"
+
+	"github.com/openshift/odo/pkg/catalog"
+	"github.com/openshift/odo/pkg/devfile"
+	"github.com/openshift/odo/pkg/log"
+	"github.com/openshift/odo/pkg/machineoutput"
+	"github.com/openshift/odo/pkg/odo/genericclioptions"
+	"github.com/openshift/odo/pkg/odo/util/experimental"
+	"github.com/openshift/odo/pkg/odo/util/pushtarget"
+	pkgUtil "github.com/openshift/odo/pkg/util"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v2"
+	"k8s.io/klog"
+	ktemplates "k8s.io/kubectl/pkg/util/templates"
+)
+
+const componentRecommendedCommandName = "component"
+
+var (
+	componentExample = ktemplates.Examples(`  # Describe a component
+    %[1]s nodejs`)
+
+	componentLongDesc = ktemplates.LongDesc(`Describe a component type.
+This describes the component and its associated starter projects.
+`)
+)
+
+// DescribeComponentOptions encapsulates the options for the odo catalog describe component command
+type DescribeComponentOptions struct {
+	// name of the component to describe, from command arguments
+	componentName string
+	// if devfile components with name that matches arg[0]
+	devfileComponents []catalog.DevfileComponentType
+	// if componentName is a classic/odov1 component
+	component string
+	// generic context options common to all commands
+	*genericclioptions.Context
+}
+
+// NewDescribeComponentOptions creates a new DescribeComponentOptions instance
+func NewDescribeComponentOptions() *DescribeComponentOptions {
+	return &DescribeComponentOptions{}
+}
+
+// Complete completes DescribeComponentOptions after they've been created
+func (o *DescribeComponentOptions) Complete(name string, cmd *cobra.Command, args []string) (err error) {
+	o.componentName = args[0]
+	if !pushtarget.IsPushTargetDocker() {
+		o.Context = genericclioptions.NewContext(cmd, true)
+		catalogList, err := catalog.ListComponents(o.Client)
```

Comment:
If you run this on a Kubernetes cluster, you will get error as it tries to list down `imageStreams` which are not present on k8s cluster. so we refrain to execute S2I related code on kubernetes cluster.

Reply:
I think not required as we are not returning error if experimental mode is enabled